_test/ensure-lib-src-rs-exist.sh

```diff
@@ -4,16 +4,46 @@ repo=$(cd "$(dirname "$0")/.." && pwd)
 
 missing=""
 
+empty_stub=""
+
+check_status=0
+
+IGNORED_EXERCISES=(
+	"two-fer" #deprecated
+	"nucleotide-codons" #deprecated
+	"hexadecimal" #deprecated
+)
+
 for dir in $repo/exercises/*/; do
-  exercise=$(basename "$dir")
-  if [ ! -f $dir/src/lib.rs ]; then
-    # https://github.com/exercism/rust/pull/270
-    echo "$exercise is missing a src/lib.rs; please create one (an empty file is acceptable)"
-    missing="$missing\n$exercise"
-  fi
+	exercise=$(basename "$dir")
+
+	if [ ! -f "$dir/src/lib.rs" ]; then
+		# https://github.com/exercism/rust/pull/270
+		echo "$exercise is missing a src/lib.rs; please create one (an empty file is acceptable)"
+		missing="$missing\n$exercise"
+	else
+		#Check if the stub file is empty
+		if [ ! -s "$dir/src/lib.rs" ] || [ "$(cat "$dir/src/lib.rs")" == "" ] && [[ " ${IGNORED_EXERCISES[*]} " != *"$exercise"* ]]; then
+			echo "$exercise has src/lib.rs stub file, but it is empty."
+			empty_stub="$empty_stub\n$exercise"
+		fi
+	fi
 done
 
 if [ -n "$missing" ]; then
-  echo "Exercises missing src/lib.rs:$missing"
-  exit 1
+	printf "\nExercises missing src/lib.rs:$missing\n"
+
+	check_status=1
+fi
+
+if [ -n "$empty_stub" ]; then
+	printf "\nExercises with empty src/lib.rs stub file:$empty_stub\n"
+
+	check_status=1
+fi
+
+if [ "$check_status" -ne 0 ]; then
```

Comment:
this could potentially be written as `exit $check_status` - was this intentionally written the proposed way for clarity, or some other reason?

Reply:
Two reasons actually:
- For clarity
- Separation of concern: `check_status` is used for internal script logic `exit 0/1` is used for external system(Travis in this case)

If this feels like unnecessary logic to you I will change it to `exit $check_status`